src/manifest.rs

```diff
@@ -45,7 +45,39 @@ fn read_cargo_toml(path: &str) -> Result<CargoManifest, Error> {
     let mut cargo_contents = String::new();
     cargo_file.read_to_string(&mut cargo_contents)?;
 
-    Ok(toml::from_str(&cargo_contents)?)
+    let contents: toml::Value = toml::from_str(&cargo_contents)?;
+    let contents = &contents["package"];
+
+    let maybe_missing_fields = vec![
+        contents.get("description").is_none(),
+        contents.get("license").is_none(),
+        contents.get("repository").is_none(),
+    ];
+
+    if maybe_missing_fields.contains(&true) {
+        let mut error_buffer = String::new();
+        if maybe_missing_fields[0] {
```

Comment:
i'm not the hugest fan of "magic numbers", i.e. these missing fields have magic numbers associated. could we do this with a data structure that has named fields?

Reply:
Like a small struct? Sure, I wasn't exactly happy about them either.